src/Evolution/Systems/GrMhd/ValenciaDivClean/FixConservatives.cpp

```diff
@@ -189,25 +189,47 @@ bool FixConservatives::operator()(
       //   converge. We handle this case explicitly because we found that
       //   setting a lower bound of 1.-1.e-12 leads to floating point
       //   exceptions.
+      // - In regions where the solution is just above atmosphere we sometimes
+      //   obtain an upper bound on the Lorentz factor somewhere around ~1e5,
+      //   while the actual Lorentz factor is only 1+1e-6. This leads to
+      //   situations where the solver must perform many (over 50) iterations to
+      //   converge. A simple way of avoiding this is to check that
+      //   10 * W_{lower_bound} brackets the root and then use that as the upper
+      //   bound. This reduces the number of iterations for the TOMS748
+      //   algorithm to converge to less than 10.
       const auto f_of_lorentz_factor = FunctionOfLorentzFactor{
           b_squared_over_d, tau_over_d, normalized_s_dot_b};
-      const double upper_bound_of_lorentz_factor = 1.0 + tau_over_d;
+      double upper_bound_of_lorentz_factor = 1.0 + tau_over_d;
 
       double lorentz_factor = std::numeric_limits<double>::signaling_NaN();
       if (const double f_at_lower =
               f_of_lorentz_factor(lower_bound_of_lorentz_factor);
-          lower_bound_of_lorentz_factor == 1.0 and f_at_lower > 0.0 and
-          f_at_lower < 1.0e-14) {
+          equal_within_roundoff(lower_bound_of_lorentz_factor,
+                                upper_bound_of_lorentz_factor) or
+          (lower_bound_of_lorentz_factor == 1.0 and f_at_lower > 0.0 and
+           f_at_lower < 1.0e-14)) {
         lorentz_factor = lower_bound_of_lorentz_factor;
       } else {
         try {
-          lorentz_factor =
-              (equal_within_roundoff(lower_bound_of_lorentz_factor,
-                                     upper_bound_of_lorentz_factor)
-                   ? lower_bound_of_lorentz_factor
-                   : RootFinder::toms748(
-                         f_of_lorentz_factor, lower_bound_of_lorentz_factor,
-                         upper_bound_of_lorentz_factor, 1.e-14, 1.e-14, 50));
+          double f_at_upper{0.0};
+          if (upper_bound_of_lorentz_factor >
+              10.0 * lower_bound_of_lorentz_factor) {
+            const double f_at_10x_lower_bound =
+                f_of_lorentz_factor(10.0 * lower_bound_of_lorentz_factor);
+            if (std::signbit(f_at_10x_lower_bound) !=
+                std::signbit(f_at_lower)) {
+              upper_bound_of_lorentz_factor =
+                  10.0 * lower_bound_of_lorentz_factor;
+              f_at_upper = f_at_10x_lower_bound;
+            }
```

Comment:
It looks like there might be a micro-optimization you can make "for free" by adding `else { f_at_lower = f_at_10x_lower_bound; }`here... this would slightly restrict the search interval and reuse a function evaluation you've already done. Unless of course if there are an even number of roots in between [lower, 10x lower]... thoughts?

Reply:
sorry, why would `f_at_lower` be set to `f_at_10x_lower_bound`? I'm only adjusting the upper bound, not the lower bound. Are you suggesting that the lower bound be adjusted since we know the root isn't in [lower, 10x lower] and so must be in [10x lower, upper]? I don't have an immediate argument against that :shrug: I think it's unlikely we will hit that case since this branch currently is only taken in atmosphere where the speed should be small. I'm happy to add if you think it's worth doing. I think it would need to be:
```cpp
} else {
  f_at_lower = f_at_10x_lower_bound;
  lower_bound_of_lorentz_factor = 10.0 * lower_bound_of_lorentz_factor;
}
```
right?

Reply:
Yes, that is what I'm suggesting, and your code snippet is the correct one, I forgot to update W_{lower} in mine! Anyway, if this is rarely relevant, maybe it's easiest to leave this as-is.

Reply:
Okay! I currently don't think we should ever hit that case, so I'm not sure it's worth adding